src/coreclr/tools/superpmi/superpmi-shim-simple/superpmi-shim-simple.cpp

```diff
@@ -14,44 +14,52 @@
 #include "spmiutil.h"
 #include "jithost.h"
 
-HMODULE g_hRealJit           = 0;       // We leak this currently (could do the proper shutdown in process_detach)
-WCHAR*  g_realJitPath        = nullptr; // We leak this (could do the proper shutdown in process_detach)
-char*   g_logFilePath        = nullptr; // We *don't* leak this, hooray!
-WCHAR*  g_HomeDirectory      = nullptr;
-WCHAR*  g_DefaultRealJitPath = nullptr;
 
-void SetDefaultPaths()
+HMODULE               g_hRealJit = 0;    // We leak this currently (could do the proper shutdown in process_detach)
+std::filesystem::path g_realJitPath{""}; // Destructable objects will be cleaned up and won't leak
+std::filesystem::path g_logPath{""};
+std::filesystem::path g_HomeDirectory{""};
+std::filesystem::path g_DefaultRealJitPath{""};
+
+// RAII holder for logger
+class LoggerHolder
```

Comment:
According to https://learn.microsoft.com/windows/win32/dlls/dllmain and other documentations I can find, constructors/destructors for global objects with be respected by dll/so unloading. I haven't test it though.